pkg/resource_deployment.go

```diff
@@ -436,9 +441,10 @@ func expandDeploymentResource(d *schema.ResourceData, defaultProject string) (*d
 			NodeDiskSize: int32(conf.nodeDiskSize),
 			NodeSizeId:   conf.nodeSizeId,
 		},
-		NotificationSettings: notificationSetting,
-		DiskAutoSizeSettings: autoSizeSettings,
-		DiskPerformanceId:    diskPerformanceID,
+		NotificationSettings:                   notificationSetting,
+		DiskAutoSizeSettings:                   autoSizeSettings,
+		DiskPerformanceId:                      diskPerformanceID,
+		IsScheduledRootPasswordRotationEnabled: true,
```

Comment:
Why not setting it to the value stored in the ResourceData, that way the HasChange will work correctly imo?

Reply:
I tried that too, was not just sure about that. because even if it's set to the `false` here, in the server it'll be `true` anyway. and we use this `expandDeployment` method on tests heavily, so with this I'm able to verify the actual behavior which is setting it to `true` even if it's `false` on creation. I don't have any strong opinion though

Reply:
Did you try to disable during creation and disable/enable during update in terraform provider, it that all works as expected I am fine.